src/firebase-shims.js

```diff
@@ -23,3 +23,18 @@ if (typeof global !== 'undefined') {
 if (XMLHttpRequest && !window.XMLHttpRequest) {
   window.XMLHttpRequest = XMLHttpRequest;
 }
+
+// WebSocket shim, currently needed for Firefox workers to use sockets
+if (typeof WebSocket === 'undefined') {
+  console.log('shimming WebSocket');
+  WebSocket = function(url, protocols) {
+    var ws = new freedom['core.websocket'](url, protocols);
+    console.log(ws);
+    this.send = ws.send;
+    this.close = ws.close;
+    ws.on('onOpen', this.onopen);
```

Comment:
I think the problem here might be that at the time this line of JavaScript is evaluated, this.onopen===undefined as opposed to the function it will be later set to.  I think this might work if you change to:
```
ws.on('onOpen', function(...) {  // possibly multiple arguments
  if (this.onopen) {
    this.onopen(..);  // same arguments
  }
}.bind(this));
```